Plugin/CategoryFieldPlugin.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace AvS\ScopeHint\Plugin;
+
+use Magento\Catalog\Api\CategoryRepositoryInterface;
+use Magento\Framework\Escaper;
+use Magento\Framework\Registry;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Ui\Component\Form\Field as FieldComponent;
+
+class CategoryFieldPlugin
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var Registry
+     */
+    protected $registry;
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    protected $categoryRepository;
+
+    /**
+     * @var Escaper
+     */
+    protected $escaper;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     * @param Registry $registry
+     * @param CategoryRepositoryInterface $categoryRepository
+     * @param Escaper $escaper
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        Registry $registry,
+        CategoryRepositoryInterface $categoryRepository,
+        Escaper $escaper
+    ) {
+        $this->storeManager = $storeManager;
+        $this->registry = $registry;
+        $this->categoryRepository = $categoryRepository;
+        $this->escaper = $escaper;
+    }
+
+    /**
+     * @param FieldComponent $subject
+     * @param $config
+     * @return array
+     */
+    public function afterGetConfiguration(FieldComponent $subject, $config)
+    {
+        if ($subject->getContext()->getNamespace() == 'category_form' &&
```

Comment:
I would go for an early return here to avoid one level of indentation :) So just directly return if not in the right context.